flaky TestBadgerBlockstore

```
Failed
=== RUN   TestBadgerBlockstore/non_prefixed/TestAllKeysRespectsContext
2020-11-11T19:35:03.875Z	�[34mINFO�[0m	badgerbs	v2@v2.2007.2/levels.go:183	All 0 tables opened in 0s

    blockstore_test_suite.go:188: 
        	Error Trace:	blockstore_test_suite.go:188
        	Error:      	Not equal: 
        	            	expected: cid.Cid{str:""}
        	            	actual  : cid.Cid{str:"\x01U\x12 \x03A& :\x9b\xaci\xd6\xed\xc0\x02~\xe9\xb3ԿsD\x995\xda\xf358gd\xc0\xd7\xf5\x12\xcf"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (cid.Cid) {
        	            	- str: (string) ""
        	            	+ str: (string) (len=36) "\x01U\x12 \x03A& :\x9b\xaci\xd6\xed\xc0\x02~\xe9\xb3ԿsD\x995\xda\xf358gd\xc0\xd7\xf5\x12\xcf"
        	            	 }
        	Test:       	TestBadgerBlockstore/non_prefixed/TestAllKeysRespectsContext
2020-11-11T19:35:03.889Z	�[34mINFO�[0m	badgerbs	v2@v2.2007.2/db.go:1030	Storing value log head: {Fid:0 Len:31 Offset:11371}

2020-11-11T19:35:03.890Z	�[34mINFO�[0m	badgerbs	v2@v2.2007.2/levels.go:1000	[Compactor: 173] Running compaction: {level:0 score:1.73 dropPrefixes:[]} for level: 0

2020-11-11T19:35:03.891Z	�[34mINFO�[0m	badgerbs	v2@v2.2007.2/levels.go:962	LOG Compact 0->1, del 1 tables, add 1 tables, took 961.211µs

2020-11-11T19:35:03.891Z	�[34mINFO�[0m	badgerbs	v2@v2.2007.2/levels.go:1010	[Compactor: 173] Compaction for level: 0 DONE
2020-11-11T19:35:03.891Z	�[34mINFO�[0m	badgerbs	v2@v2.2007.2/db.go:553	Force compaction on level 0 done
        --- FAIL: TestBadgerBlockstore/non_prefixed/TestAllKeysRespectsContext (0.04s)
```

--- a/lib/blockstore/badger/blockstore.go
+++ b/lib/blockstore/badger/blockstore.go
@@ -359,7 +359,11 @@ func (b *Blockstore) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
 				buf = make([]byte, reqlen)
 			}
 			if n, err := base32.RawStdEncoding.Decode(buf, k); err == nil {
-				ch <- cid.NewCidV1(cid.Raw, buf[:n])
+				select {
+				case ch <- cid.NewCidV1(cid.Raw, buf[:n]):
+				case <-ctx.Done():
+					return
+				}
 			} else {
 				log.Warnf("failed to decode key %s in badger AllKeysChan; err: %s", k, err)
 			}

--- a/lib/blockstore/badger/blockstore_test_suite.go
+++ b/lib/blockstore/badger/blockstore_test_suite.go
@@ -219,6 +219,9 @@ func (s *Suite) TestReopenPutGet(t *testing.T) {
 	fetched, err := bs.Get(orig.Cid())
 	require.NoError(t, err)
 	require.Equal(t, orig.RawData(), fetched.RawData())
+
+	err = bs.(io.Closer).Close()
+	require.NoError(t, err)
 }
 
 func (s *Suite) TestPutMany(t *testing.T) {

